airflow/jobs/triggerer_job_runner.py

```diff
@@ -608,6 +609,14 @@ async def run_trigger(self, trigger_id, trigger):
                 self.log.info("Trigger %s fired: %s", self.triggers[trigger_id]["name"], event)
                 self.triggers[trigger_id]["events"] += 1
                 self.events.append((trigger_id, event))
+
+            trigger_timeout = trigger.task_instance.trigger_timeout
+            if trigger_timeout and trigger_timeout < timezone.utcnow():
+                self.log.info("Trigger cancelled due to timeout")
+                raise asyncio.CancelledError("Trigger cancelled due to timeout")
```

Comment:
i think this needs to be moved inside the `except asyncio.CancelledError as err` block

Reply:
it's pretty easy to test you can use example_time_delta_sensor_async.py and temporarily set its defer timeout to 2 seconds or something.

Reply:
Hi, thanks for the suggestions for the changes and how to test them. I've made the changes accordingly and tested it locally.